app/lib/apicast/provider_source.rb

```diff
@@ -46,7 +46,7 @@ def attributes_for_proxy
         ]
       }
 
-      ActiveRecord::Associations::Preloader.new.preload(provider,  {services: [:service_tokens, {backend_api_configs: :backend_api, proxy: [:gateway_configuration, {proxy_rules: :metric}]}]})
+      ActiveRecord::Associations::Preloader.new(records: Array(provider), associations: {services: [:service_tokens, {backend_api_configs: :backend_api, proxy: [:gateway_configuration, {proxy_rules: :metric}]}]})
```

Comment:
This was required after this refactoring: https://github.com/rails/rails/pull/40776